pkg/controllers/machinesync/aws_test.go

```diff
@@ -291,3 +292,126 @@ var _ = Describe("AWS load balancer validation during MAPI->CAPI conversion", fu
 		Eventually(k8sClient.Get(ctx, client.ObjectKeyFromObject(capiMachine), capiMachine), timeout).Should(Succeed())
 	})
 })
+
+var _ = Describe("validateLoadBalancerReferencesAgainstExpected", func() {
+	var (
+		lbfieldPath *field.Path
+	)
+
+	BeforeEach(func() {
+		lbfieldPath = field.NewPath("spec", "providerSpec", "value", "loadBalancers")
+	})
+
+	type validateLoadBalancerMatchTableInput struct {
+		actualLoadBalancers   []mapiv1beta1.LoadBalancerReference
+		expectedLoadBalancers map[string]mapiv1beta1.AWSLoadBalancerType
+		expectError           bool
+		expectedErrorMessages []string
+	}
+
+	DescribeTable("validate load balancer matching",
+		func(in validateLoadBalancerMatchTableInput) {
+			err := validateLoadBalancerReferencesAgainstExpected(in.actualLoadBalancers, in.expectedLoadBalancers, lbfieldPath)
+
+			if in.expectError {
+				Expect(err).ToNot(BeNil())
```

Comment:
```suggestion
			if len(in.expectedErrorMessages) {
```

So we can get rid of the `expectError` parameter.